cli/src/commands/serve_web.rs

```diff
@@ -568,6 +569,25 @@ impl ConnectionManager {
 			return Ok(previous.clone());
 		}
 
+		// If the update service is unavailable and we have cached data, use that
+		if let Err(e) = &release {
```

Comment:
I think it'd be easier to seed `self.latest_version` from the LRU in `ConnectionManager::new`, then the conditional above would take care of this case

Reply:
I have also observed that when stale data is being used the conditional above does not set the latest, which intern results in the skipping of the cache reuse conditional. During a situation like no internet where the `get_latest_commit` will fail, each new connection ends up checking for the latest commit which fails after a few seconds and return the previously know latest release. Should we set the `latest` with the current timestamp so that new connection can get currently available release without having to wait for a few seconds for the check to fail?

Reply:
I would set `latest` with a time of 0 so that it tries to ask the update service the first time.

Currently the `latest_version` contains the Instant the version was last checked at. I would change this to instead represent the Instant the next check _should happen at_ and increment it by a minute or so each time the request fails.

Reply:
@connor4312 I have separated the update check, it now checks for updated independently and duration can be adjusted using the flag `update_check_interval` which can also be used to turn off the check with a `0` value. The server starts with the latest version stored in cache instead of checking for updated every time.